modules/effects/src/effects_resolver.ts

```diff
@@ -1,38 +1,50 @@
 import { Action } from '@ngrx/store';
 import { merge, Notification, Observable } from 'rxjs';
-import { ignoreElements, map, materialize, catchError } from 'rxjs/operators';
+import {
+  ignoreElements,
+  map,
+  materialize,
+  catchError,
+  tap,
+} from 'rxjs/operators';
 
 import { EffectNotification } from './effect_notification';
 import { getSourceMetadata } from './effects_metadata';
 import { getSourceForInstance } from './utils';
 import { ErrorHandler } from '@angular/core';
 
+export type EffectsErrorHandler = <T extends Action>(
+  observable$: Observable<T>,
+  errorHandler?: ErrorHandler
```

Comment:
we should probably make it not-optional:
- it would be harder to add addition _required_ arguments (should our custom error handling functions have those)
- it's present Angular and passed into `mergeEffects` anyway

Reply:
agree, done